enhancements/ingress/optional-ingress-hypershift.md

```diff
@@ -0,0 +1,272 @@
+
+## Motivation
+
+Users of the ROSA managed OpenShift service would like to use [AWS load balancers](/enhancements/ingress/aws-load-balancer-operator.md) as a way to accept the ingress traffic into the cluster.
+The cluster ingress operator becomes unnecessary and can be disabled to:
+- reduce the resource consumption on
+  - the HyperShift client clusters (ingress controllers)
+  - the HyperShift hosted control plane (operator)
+- make the HyperShift hosted control plane lighter, simpler therefore more manageable
+- enable the HyperShift cluster users to make their choice for the cluster ingress
+
+### Goals
+- Implement the new ingress capability.
+- Add the ingress capability to the ingress operator's payload.
+- Make the dependent cluster operators tolerate the absence of the ingress operator.
+- Do not lock the subsequent implementation for the standalone OpenShift.
+
+### Non-Goals
+- Use a different way for enabling/disabling the ingress other than via [the cluster capability](/enhancements/installer/component-selection.md).
+- Fully implement the ingress capability for the standalone OpenShift:
+  - Make the OpenShift installer tolerate the absence of the default ingress controller.
+  - Make the cluster authentication operator tolerate the absence of the default ingress controller.
+  - Make the cluster monitoring operator tolerate the absence of the default ingress controller.
+- Disable the Route API or try to tolerate the absence of the default ingress controller.
+- Design the capability API for HyperShift.
+- Implement the ingress operator disabling in the HyperShift's control plane operator.
+
+## Proposal
+
+### User Stories
+
+- As a HyperShift engineer, I want to be able to disable the ingress operator on the clusters where the ingress is implemented differently so that the hosted control plane can become lighter.
+- As a HyperShift engineer, I want the cluster operators to tolerate the absence of the ingress operator to keep the clusters from being degraded.
+- As a HyperShift hosted cluster's user, I want to see the ingress capability as a known capability to bring the visibility to all the cluster actors (admins, users, other operators).
+- As a HyperShift hosted cluster's user, I want the default ingress controller to not occupy the cluster resources while the ingress traffic is routed by the cloud provider's load balancers.
+
+### Workflow Description
+
+The HyperShift hosted cluster's user wants to use an AWS ALB to route the traffic to its example application deployed on a HyperShift's hosted cluster.
+
+#### Terminology
+[HyperShift concepts and personas](https://hypershift-docs.netlify.app/reference/concepts-and-personas/).
+
+#### Disable the default ingress
+
+- Cluster service consumer requests a hosted cluster without the ingress capability using the HostedCluster API (**note**: the HostedCluster's capability API does not exist and it's out of scope of this enhancement).
```

Comment:
I suppose it'll be the same API as for the cluster version or the install config, so no way to disable explicitly, just by using the `None` baseline set. However I cannot really speak for the HyperShift engineers here.

Reply:
i'm still a little confused about the mechanics overall i think, but maybe when you address my larger question from the review it will be clearer how the pieces work together.